mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl

```diff
@@ -541,7 +541,8 @@ func ResourceComputeInstance() *schema.Resource {
 
 						"ipv6_access_config": {
 							Type:        schema.TypeList,
-                                                        Optional:    true,
+                            Optional:    true,
+							Computed:    true,
```

Comment:
@karolgorc , @ChuAmberZhang , adding `Computed: true` will suppress the diffs. But it will prevent the removal of `ipv6_access_config` for users, as the previous value will be used. Do you think it will affect a lot of current users with `IPV4_IPV6` and `IP4_ONLY` stackType as `ipv6_access_config` cannot be removed?

Reply:
There is not a way to make the removal of `ipv6_access_config` work with adding `Computed: true` with the current Terraform SDK library.

Reply:
Another way to suppress the diffs is adding a diff suppress function for `IPV4_IPV6` and `IPV6_ONLY` stackType and to not add `Computed: true`. The function will also prevent the removal of `ipv6_access_config`.

Reply:
Or it is OK for the users to see the diffs now?  Or we can just suppress the diffs for the new `IPV6_ONLY` stacktype?

Reply:
> @karolgorc , @ChuAmberZhang , adding `Computed: true` will suppress the diffs. But it will prevent the removal of `ipv6_access_config` for users, as the previous value will be used. Do you think it will affect a lot of current users with `IPV4_IPV6` and `IP4_ONLY` stackType as `ipv6_access_config` cannot be removed? Based on my understanding, `google_compute_subnetwork`, `google_compute_instance`, `google_compute_instance_template` and `google_compute_region_instance_template` are widely used resources.

Hi @zli82016 do you mean the field can't be removed from the request input or output? There are a few cases to consider:

1. This field is not supposed to be populated for `IPV4_ONLY` stack type interfaces and `IPV4_IPV6` stack type interfaces with `INTERNAL `IPv6 access type, and `IPV6_ONLY` stack type interfaces with `INTERNAL` IPv6 access type
2. `IPV4_IPV6` and `IPV6_ONLY` interfaces with `EXTERNAL` IPv6 access type will have this field populated and this field should not be modifiable by the user (or any change attempt will be ignored or error out). When the stack type is updated to `IPV4_ONLY`, the field will be cleared automatically. 
3. When creating new `IPV4_IPV6` or `IPV6_ONLY` interfaces or changing the stack type from `IPV4_ONLY` to `IPV4_IPV6`, users should be able to specify this field in the request.
4. For a single instance, different interfaces may have different stack types and IPv6 access type, and thus may have different `ipv6_access_config` fields.

It's quite complicated, so we need to make sure if we set `Compute: true` the above cases are still honored.

Reply:
With `Compute: true`, even this field is removed from the resource configuration, this field will still be sent to the API with the field's previous value. 

`Computed: true` will be added always for all of the cases.  There is no way to only set this field computed in some cases.

Reply:
Another option is to add the [DiffSuppressFunc](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl#L263) to this field to only suppress diffs for the case: IPV4_IPV6 and IPV6_ONLY interfaces with EXTERNAL IPv6 access type . This function will suppress the diffs between the empty value in resource configuration (new value) and the default value stored in Terraform state (old value). 

There are two issues with this option:
1. When this field is removed from the resource configuration, the diffs are suppressed by this function. It is the same issue with `Compute: true`. 
2. I doubt if there is a way to know IPv6 access type is EXTERNAL

Reply:
This can't be diff suppressed. The info on whether the subnetwork is `External` or `Internal` isn't held by the Instance. It's held by the subnetwork. 

Doing API calls on external resources to suppress a diff seems like a bad design to me.

We can add a condition to the flattening function that will set this field to null unless it's specified by the user in his config. Making this show no diff but it will still get the right values from the API. Something like [this](https://github.com/GoogleCloudPlatform/magic-modules/pull/12552). I will have to test how overriding this works.

> But it will prevent the removal of ipv6_access_config for users, as the previous value will be used.

I don't think so. If the user specifies it he can just update this field. But if not i think it's expected that ipv6_access_config is set automatically when the instance is given a subnetwork with external addresses.

Reply:
> > But it will prevent the removal of ipv6_access_config for users, as the previous value will be used.
> 
> I don't think so. If the user specifies it he can just update this field. But if not i think it's expected that ipv6_access_config is set automatically when the instance is given a subnetwork with external addresses.
> 

With the `Computed: true + Optional: true`, when the user wants to remove `ipv6_access_config` from the configuration, its previous value from the Terraform state will be used to be sent to API. That is how `Computed: true + Optional: true` works.

Reply:
> To sum things up i think that having this as `Computed: true` is a must. Yeah we can just do diff suppress instead of this but this will restrict API's functionality. The user will not be able to query for essential values like the IP on this access config.

Adding `Compute: true` makes it impossible to remove `ipv6_access_config` (setting it to nil from Terraform), even `ipv6_access_config` is removed from the resource configuration. It will affect all of the stack types. I don't think that is what we want.

Reply:
Do you think that the diffs must be suppressed now? Do we consider leaving the diffs right now? I wonder if you have received  a lot of issues related to the diffs for `IPV4_IPV6` stack type?

Reply:
Not sure if leaving the diff is optimal when shipping this feature. But if you think that removing the `Computed`, adding tests, and maybe a doc update to inform about this is enough. They yeah, for sure.

Reply:
We can leave the diffs and remove the `Computed` now. You can create a Github issue with this [link](https://github.com/hashicorp/terraform-provider-google/issues/new?assignees=&labels=bug&projects=&template=00_bug.yml) for the diffs